src/kafka/kafka.ts

```
@@ -0,0 +1,109 @@
+
+  constructor(ctor: CT) {
+    super(ctor);
+  }
+}
+
+export function Kafka(kafkaConfig: KafkaConfig) {
+  function clsdec<T extends { new(...args: unknown[]): object }>(ctor: T) {
+    const clsreg = getOrCreateClassRegistration(ctor) as KafkaClassRegistration<T>;
+    clsreg.kafkaConfig = kafkaConfig;
+  }
+  return clsdec;
+}
+
+////////////////////////
+/* Kafka Management  */
+///////////////////////
+
+export class DBOSKafka{
+  readonly consumers: Consumer[] = [];
+
+  constructor(readonly dbosExec: DBOSExecutor) {}
+
+  async initKafka() {
+    for (const registeredOperation of this.dbosExec.registeredOperations) {
+      const ro = registeredOperation as KafkaRegistration<unknown, unknown[], unknown>;
+      if (ro.topic) {
+        const defaults = ro.defaults as KafkaDefaults;
+        const kafka = new KafkaJS(defaults.kafkaConfig!);
+        const consumerConfig = ro.consumerConfig ?? { groupId: `dbos-kafka-group-${ro.topic}`}
+        const consumer = kafka.consumer(consumerConfig);
+        await consumer.connect()
+        await consumer.subscribe({topic: ro.topic, fromBeginning: true})
+        await consumer.run({
+          eachMessage: async ({ topic, partition, message }) => {
+            // This combination uniquely identifies a message for a given Kafka cluster
+            const workflowUUID = `kafka-unique-id-${topic}-${partition}-${message.offset}`
+            const wfParams = { workflowUUID: workflowUUID };
+            // All operations annotated with Kafka decorators must take in these three arguments
+            const args: KafkaArgs = [topic, partition, message]
+            // We can only guarantee exactly-once-per-message execution of transactions and workflows.
+            if (ro.txnConfig) {
+              // Execute the transaction
+              await this.dbosExec.transaction(ro.registeredFunction as Transaction<unknown[], unknown>, wfParams, ...args);
+            } else if (ro.workflowConfig) {
```

Log/throw an error if the config is something else? The current code seems to silently ignore communicators or middleware decorated with a KafkaConsumer decorator.